### PR TITLE
dmc walker

### PR DESCRIPTION
**Pull request by yufansong**
#103

### CODE REVIEW

File: envpool/mujoco/dmc/walker.h [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+class WalkerEnv : public Env<WalkerEnvSpec>, public MujocoEnv {
+ protected:
+  // Minimal height of torso over foot above which stand reward is 1.
+  const mjtNum kStandHeight = 1.2;
+  // Horizontal speeds(meters / second) above which move reward is 1.
+  const mjtNum kWalkSpeed = 1;
+  const mjtNum kRunSpeed = 8;
+  mjtNum* kOrient = nullptr;
+  float moveSpeed_;
```
Comment (Trinkle23897) [inline envpool/mujoco/dmc/walker.h:75]:
```suggestion
  float move_speed_;
```

File: envpool/mujoco/dmc/mujoco_dmc_align_test.py [hunk 2]
```diff
@@ -87,7 +89,7 @@ def run_align_check(
         for k in obs_spec:
           np.testing.assert_allclose(o0[k], getattr(o1, k)[0])
         np.testing.assert_allclose(ts0.step_type, ts1.step_type[0])
-        np.testing.assert_allclose(ts0.reward, ts1.reward[0])
+        np.testing.assert_allclose(ts0.reward, ts1.reward[0], rtol=1e-6)
```
Comment (yufansong) [inline envpool/mujoco/dmc/mujoco_dmc_align_test.py:92]: If I do not change the rtol from 1e-7 to 1e-6, the reward will fail. I am not sure if there is a bug in my code or if there exists a computation accuracy problem.

File: envpool/mujoco/dmc/walker.h [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  float move_speed_;
+
+ public:
+  WalkerEnv(const Spec& spec, int env_id)
+      : Env<WalkerEnvSpec>(spec, env_id),
+        MujocoEnv(
+            spec.config["base_path"_],
+            GetWalkerXML(spec.config["base_path"_], spec.config["task_name"_]),
+            spec.config["frame_skip"_], spec.config["max_episode_steps"_]) {
+    k_orient_ = static_cast<mjtNum*>(std::malloc(14 * sizeof(mjtNum)));
```
Comment (Trinkle23897) [inline envpool/mujoco/dmc/walker.h:80]: don't use malloc and free in modern c++, instead, use `std::unique_ptr`

File: envpool/mujoco/dmc/walker.h [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  }
+  mjtNum TorsoHeight() {
+    //   return self.named.data.xpos['torso', 'z']
+    return data_->xpos[1 * 3 + 2];
+  }
+  mjtNum HorizontalVelocity() {
+    // return self.named.data.sensordata['torso_subtreelinvel'][0]
+    return data_->sensordata[0];
+  }
+  mjtNum* Orientations() {
```
Comment (Trinkle23897) [inline envpool/mujoco/dmc/walker.h:161]: so why not directly return `std::array<mjtNum, 14>`?

File: envpool/mujoco/dmc/walker.h [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  }
+
+  void Step(const Action& action) override {
+    mjtNum* act = static_cast<mjtNum*>(action["action"_].Data());
+    ControlStep(act);
+    WriteState();
+  }
+
+  float TaskGetReward() override {
+    auto standing = static_cast<float>(RewardTolerance(
```
Comment (Trinkle23897) [inline envpool/mujoco/dmc/walker.h:118]: static_cast in return would be more elegant?

**Review by Trinkle23897 — DISMISSED** [review]

**Review by Trinkle23897 — APPROVED** [review]

**Pull request merged**